posydon/popsyn/synthetic_population.py

```diff
@@ -992,16 +992,38 @@ def __init__(
                 Pwarn(f"{filename} already contains a mass_per_metallicity "
                       "table. Overwriting the table!", "OverwriteWarning")
 
-            simulated_mass = np.sum(self.oneline[["S1_mass_i", "S2_mass_i"]].to_numpy())
+            try:
+        # Filter the data based on the condition
+                filtered_data = self.oneline[self.oneline["state_i"] == "initially_single_star"]
+                filtered_data1 = self.oneline[self.oneline["state_i"] != "initially_single_star"]
+    
+        # Calculate the simulated masses
+                simulated_mass_binaries = np.sum(filtered_data1[["S1_mass_i", "S2_mass_i"]].to_numpy())
+                simulated_mass_single = np.sum(filtered_data[["S1_mass_i", "S2_mass_i"]].to_numpy())
+                simulated_mass = np.sum(self.oneline[["S1_mass_i", "S2_mass_i"]].to_numpy())
+    
+            except TypeError as e:
+                #print(f"KeyError: {e}. Check if the specified columns exist in the DataFrame.")
+                
+                filtered_data = None
+                filtered_data1 = self.oneline[(self.oneline["state_i"] != "initially_single_star")]
+                simulated_mass_binaries = np.sum(filtered_data1[["S1_mass_i", "S2_mass_i"]].to_numpy())
+                simulated_mass_single = None
+                simulated_mass = np.sum(self.oneline[["S1_mass_i", "S2_mass_i"]].to_numpy())
+                print(1)
+            
             underlying_mass = initial_total_underlying_mass(
-                df=simulated_mass, **self.ini_params
-            )[0]
+            df=simulated_mass, df1=simulated_mass_single, df2= simulated_mass_binaries, **self.ini_params
```

Comment:
According to the function's doc-string `df` should be a pandas data frame, but you pass here a float.